app/src/main/java/com/example/egida/presentation/ui/SettingFragment.kt

```diff
@@ -22,20 +23,109 @@ class SettingFragment : Fragment() {
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect { userDatabase ->
+                        userDatabase.lastName = editable.toString()
+                        Log.d(TAG, userDatabase.lastName)
+                    }
+            }
+        }
+        binding.checkAgreement.setOnClickListener {
+            if (binding.checkAgreement.isChecked) {
+                lifecycleScope.launch {
+                    settingViewModel.userDatabase
+                        .collect { userDatabase ->
+                            userDatabase.checkAgreement = true
+                            Log.d(TAG, userDatabase.checkAgreement.toString())
+                        }
+                }
+            }
+        }
+        binding.editPhoneNumber.doAfterTextChanged { editable ->
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect { userDatabase ->
+                        userDatabase.phoneNumber = editable.toString()
+                        Log.d(TAG, userDatabase.phoneNumber)
+                    }
+            }
+        }
+        binding.addPhoto.setOnClickListener {
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect { userDatabase ->
+                        userDatabase.photoURL = it.toString()
+                        Log.d(TAG, userDatabase.photoURL)
+                    }
+            }
+        }
+        binding.minusHeight.setOnClickListener {
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect {
+                        it.height--
+                        binding.textHeight.text = it.height.toString()
+                    }
+            }
+        }
+        binding.plusHeight.setOnClickListener {
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect {
+                        it.height++
+                        binding.textHeight.text = it.height.toString()
+                    }
+            }
+        }
+        binding.minusWeight.setOnClickListener {
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect {
+                        it.weight--
+                        binding.textWeight.text = it.weight.toString()
+                    }
+            }
+        }
+        binding.plusWeight.setOnClickListener {
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect {
+                        it.weight++
+                        binding.textWeight.text = it.weight.toString()
+                    }
+            }
+        }
+
+        binding.settingsFragmentBtnSave.setOnClickListener {
+            lifecycleScope.launch {
+                settingViewModel.save()
+            }
```

Comment:
No need in starting a new coroutine here. Just call viewModel method. 'save' method should start new coroutine.